packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -491,6 +491,22 @@ class FlutterVmService {
   final Uri? wsAddress;
   final Uri? httpAddress;
 
+  /// Calls [service.getVM]. However, in the case that an [vm_service.RPCError]
+  /// is thrown due to the service being disconnected, the error is discarded
+  /// and null is returned.
+  Future<vm_service.VM?> getVmGuarded() async {
+    try {
+      return await service.getVM();
+    } on vm_service.RPCError catch (err) {
+      if (err.code == RPCErrorCodes.kServiceDisappeared ||
+          err.message.contains('Service connection disposed')) {
```

Comment:
I'm aware of https://github.com/dart-lang/sdk/commit/e099a00ee263460a745f3bad9e51535d98d87ff7, but I don't think the new error code is exposed anywhere yet.

Reply:
Yeah, it looks like we'll need to publish a new version of `package:vm_service`, but if we're going to be CP'ing this change this is probably sufficient for now.